src/mintpy/prep_hyp3.py

```diff
@@ -26,19 +26,29 @@ def add_hyp3_metadata(fname, meta, is_ifg=True):
         Metadata dictionary (meta)
     '''
 
-    # determine interferogram pair info and hyp3 metadata file name
-    sat, date1_str, date2_str, pol, res, soft, proc, ids, *_ = os.path.basename(fname).split('_')
-    job_id = '_'.join([sat, date1_str, date2_str, pol, res, soft, proc, ids])
-    directory = os.path.dirname(fname)
-    meta_file = f'{os.path.join(directory,job_id)}.txt'
-
-    # open and read hyp3 metadata
+    # read hyp3 metadata file
+    # e.g.: burst-wide product using ISCE2: {SAT}_{FRAME}_{SUBSWATH}_{DATE1}_{DATE2}_{POL}_{RES}_{IDS}.txt
+    #       scene-wide product using Gamma: {SAT}_{DATE1}_{DATE2}_{POL}_{RES}_{SOFT}_{PROC}_{IDS}.txt
+    job_id = '_'.join(os.path.basename(fname).split('_')[:8])
+    meta_file = os.path.join(os.path.dirname(fname), f'{job_id}.txt')
     hyp3_meta = {}
     with open(meta_file) as f:
         for line in f:
             key, value = line.strip().replace(' ','').split(':')[:2]
             hyp3_meta[key] = value
 
+    # get date1/2 objects
+    if job_id.split('_')[2].startswith('IW'):
```

Comment:
I modified how we identify the burst product by checking the `IW` character, instead of [counting the segment number](https://github.com/insarlab/MintPy/pull/1074/commits/33935b632cc2055f28f722a14f1012cf49acc6f7#diff-fc1cc7ebaad751500fc617a19199049f496a17a0e5ae8868c4b2acaf89bf848aR30), in the filename, hoping it to be more robust. Does this make sense to you? @cirrusasf @forrestfwilliams

Reply:
Yes. This make more sense. Thank you.

Reply:
I prefer this version as well, thanks Yunjun!